src/main/java/com/google/sps/servlets/NaturalLanguageServlet.java

```diff
@@ -63,9 +95,29 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         if (this.nlp == null) {
             this.nlp = new NaturalLanguageProcessor();
         }
+        if (this.db == null) {
+            this.db = new DatabaseImpl();
+        }
 
         // Builds the Java object from JSON and preprocesses the captions by redefining time ranges
         YoutubeCaptions youtubeCaptions = gson.fromJson(json, YoutubeCaptions.class);
+
+        // Parses the video ID from the url
+        boolean addToDatabase = true;
+        String videoID = "";
+        String[] urlParts = youtubeCaptions.getVideoURL().split(VIDEO_URL_ID_DELIMITER);
+        if (urlParts.length <= 1) {
+            addToDatabase = false;
+        }
+        if (addToDatabase) {
+            videoID = urlParts[1];
+            int ampersandPosition = videoID.indexOf(URL_QUERY_DELIMITER);
+            if (ampersandPosition != -1) {
+                videoID = videoID.substring(0, ampersandPosition);
+            }
+            db.addVideo(videoID, DB_NO_METADATA);
+        }
+
         int numCaptions = youtubeCaptions.getCaptions().size();
```

Comment:
I believe it is better to move
```
        NaturalLanguagePreprocessor preprocessor = new NaturalLanguagePreprocessor();
        List<TimeRangedText> preprocessedResults = preprocessor.setTimeRanges(youtubeCaptions.getCaptions());
        
        // Sends the text of newly defined time ranges to the NLP API and organizes the results in the postprocessor
        NaturalLanguagePostprocessor postprocessor = new NaturalLanguagePostprocessor();
        for (TimeRangedText text : preprocessedResults) {
            postprocessor.addEntities(nlp.getEntities(text.getText()), text.getStartTime());
        }
        Map<String, List<Long>> resultMap = postprocessor.getEntitiesMap();
```
to a new function which returns Map<String, List<Long>> . The post function is a little too long.

Reply:
addressed